### ISSUE TITLE
Is it possible to configure the gateway timeout ?

### ISSUE THREAD

**Issue by eltonio450** [awslabs/aws-mobile-appsync-sdk-js#108]
Hello :),

I have a lambda resolver that takes ~40 s to be executed (1 min if cold start), and I did not see where to configure the appsync timeout.

When I try to request this ressource, I get a 504 error after ~30s with x-cache: Error from cloudfront

Is there a way to set up the cloudfront timeout somewhere (via the SDK, a specific header or a configuration file) ?

Thank you

**Comment by eltonio450** [comment]
Hi,

Thank you for the answer. It may be a problem in the long run: a GraphQL API is supposed to be one unique endpoint, and it's one of its strengths.

But some tasks are a bit longer than 30s (in our case: Machine Learning prediction on Lambda functions, FTP file transfers/EDI for example). We saw a real advantage to centralize all our calls with GraphQL (Appsync) + Cognito. The goal is to have a common API for all our services accessible from our ERP.

Do you think it could change in the future ?

Best regards,

**Comment by ryanwmarsh** [comment]
@eltonio450 might I recommend returning the calls immediately and then firing a subscription event (mutation) when the processing is complete? That way the client will be notified when processing is complete and it can take as long as necessary?

**Comment by eltonio450** [comment]
@ryanwmarsh thanks for the suggestion :). I am not sure to fully understand: 

Do you mean that the lambda (for example) should implement a graphQL client too (or at least send a minimal request to the endpoint, including authentication :( ) and trigger subscription events ?

**Comment by ryanwmarsh** [comment]
@eltonio450 Yes I think you have it right. You could have the client make a graphql mutation (or just a regular REST call to the lambda) which kicks off the processing. When processing completes you could have the lambda connect to the graphql API and kick off a mutation (perhaps with simply a local resolver). If you have a subscription for said mutation then the client can subscribe and be notified when processing is complete. 

I admit it isn't pretty but mutation subscriptions and local resolvers are quite powerful.

**Comment by eltonio450** [comment]
Thanks ! I'll give a try 😃
